packages/react-aria-components/src/GridList.tsx

```diff
@@ -80,14 +90,33 @@ interface GridListInnerProps<T extends object> {
 }
 
 function GridListInner<T extends object>({props, collection, gridListRef: ref}: GridListInnerProps<T>) {
-  let {dragAndDropHooks} = props;
+  let {dragAndDropHooks, navigationBehavior = 'arrow', layout = 'stack'} = props;
   let state = useListState({
     ...props,
     collection,
     children: undefined
   });
 
-  let {gridProps} = useGridList(props, state, ref);
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let {disabledBehavior, disabledKeys} = state.selectionManager;
+  let keyboardDelegate = useMemo(() => (
+    new ListKeyboardDelegate({
+      collection,
+      collator,
+      ref,
+      disabledKeys,
+      disabledBehavior,
+      layout,
+      direction: 'ltr'
+    })
+  ), [collection, ref, layout, disabledKeys, disabledBehavior, collator]);
+
+  let {gridProps} = useGridList({
+    ...props,
+    keyboardDelegate,
+    // Only tab navigation is supported in grid layout.
+    navigationBehavior: layout === 'grid' ? 'tab' : navigationBehavior
```

Comment:
confusing? I originally only had `layout` as a prop, but realized that `navigationBehavior="tab"` might be useful for some cases even in stack layout so now it's two props. But I don't think arrow navigation in grid layout doesn't really make sense? But maybe? I guess table does that...